3p/integration.js

```diff
@@ -117,6 +117,9 @@ const AMP_EMBED_ALLOWED = {
 // used for extracting fakead3p from production code.
 const IS_DEV = true;
 
+const data = parseFragment(location.hash);
```

Comment:
https://github.com/ampproject/amphtml/blob/master/src/3p-frame.js#L114
It will always be valid JSON if the iframe is created during runtime. I am not sure about testing